src/main/java/com/spikes2212/dashboard/AutoChooser.java

```diff
@@ -0,0 +1,159 @@
+        this(rootNamespaceName, defaultOption, defaultOptionName, 4, options);
+    }
+
+    /**
+     * Creates an autonomous chooser where the first {@link Command} and {@link String} will be used as the default
+     * option, and the rest of the options will be parsed as following: every even index (0,2,4,...) will be a command
+     * and every odd index (1,3,5,...) will be used as the name for the previous command.
+     *
+     * @param defaultOption the default command this {@link SendableChooser} will use as the default option
+     * @param defaultOptionName the name for the default command
+     * @param options the rest of the options, where every even index is a command to be added and every odd index is
+     *               the name for the previous command
+     */
+    public AutoChooser(Command defaultOption, String defaultOptionName, Object... options) {
+        this(DEFAULT_NAMESPACE_NAME, defaultOption, defaultOptionName, 3, options);
+    }
+
+    private AutoChooser(String rootNamespaceName, Command defaultOption, String defaultOptionName,
+                        int incrementationValue, Object... options) {
+        rootNamespace = new RootNamespace(rootNamespaceName);
+        setDefaultOption(defaultOptionName, defaultOption);
+        Command command;
+        String name;
+        int num;
+        for (int i = 0; i < options.length - options.length % 2; i += 2) {
+            if (options[i] instanceof Command) {
+                command = (Command) options[i];
+            } else {
+                num = i + incrementationValue;
+                throw new IllegalArgumentException("The " + num + getCorrectSuffix(num) + " argument is not a Command.");
+            }
+            if (options[i + 1] instanceof String) {
+                name = (String) options[i + 1];
+            } else {
+                num = i + incrementationValue + 1;
+                throw new IllegalArgumentException("The " + num + getCorrectSuffix(num) + " argument is not a String.");
+            }
+            addOption(name, command);
+        }
+        putOnShuffleboard();
+    }
+
+    /**
+     * Adds the given command with the given name as an option to this autonomous chooser. <br>
+     * In case this name already exists, a number will be added to differentiate between the options.
+     */
+    @Override
+    public void addOption(String name, Command command) {
+        if (!addName(name)) {
+            int i = 2;
+            name += " " + i;
+            while (!addName(name)) {
+                name = name.substring(0, name.length() - String.valueOf(i).length());
+            }
+        }
+        super.addOption(name, command);
+    }
+
+    /**
+     * Schedules the selected command and closes this {@link SendableChooser}.
+     */
+    public void schedule() {
+        getSelected().schedule();
+        close();
+    }
+
+    private String getCorrectSuffix(int num) {
```

Comment:
This method does not always return the correct suffix. For example, 21 should become 21st, but this method will return "th".